--- /\net\minecraft\world\chunk
+++ /\net\minecraft\world\chunk
@@ -409,6 +409,10 @@
 
    @Nullable
    public IBlockState setBlockState(BlockPos var1, IBlockState var2) {
+      //OML
+      xyz.openmodloader.event.impl.BlockEvent.PlaceEvent event = new xyz.openmodloader.event.impl.BlockEvent.PlaceEvent(this.worldObj, var2, var1);
+      if (!xyz.openmodloader.event.Events.BLOCK_PLACE.post(event)) return var2; else var2 = event.getBlockState();
+      //OML END
       int var3 = var1.getX() & 15;
       int var4 = var1.getY();
       int var5 = var1.getZ() & 15;
