--- /net/minecraft/client/audio
+++ /net/minecraft/client/audio
@@ -69,7 +69,7 @@
          try {
             for(IResource var6 : var1.getAllResources(new ResourceLocation(var3, "sounds.json"))) {
                try {
-                  Map var7 = this.getSoundMap(var6.getInputStream());
+                  Map<String, SoundList> var7 = this.getSoundMap(var6.getInputStream());
 
                   for(Entry var9 : var7.entrySet()) {
                      this.loadSoundResource(new ResourceLocation(var3, (String)var9.getKey()), (SoundList)var9.getValue());
