--- /net/minecraft/client
+++ /net/minecraft/client
@@ -40,8 +40,6 @@
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
-import net.minecraft.client.ClientBrandRetriever;
-import net.minecraft.client.LoadingScreenRenderer;
 import net.minecraft.client.audio.MusicTicker;
 import net.minecraft.client.audio.SoundHandler;
 import net.minecraft.client.entity.EntityPlayerSP;
@@ -389,6 +387,7 @@
    }
 
    private void startGame() throws LWJGLException {
+      xyz.openmodloader.OpenModLoader.INSTANCE.minecraftConstruction();
       this.gameSettings = new GameSettings(this, this.mcDataDir);
       this.defaultResourcePacks.add(this.mcDefaultResourcePack);
       this.startTimerHackThread();
@@ -787,6 +786,10 @@
    }
 
    public void displayGuiScreen(@Nullable GuiScreen var1) {
+//    OML START
+      xyz.openmodloader.event.impl.GuiEvent.Open event = new xyz.openmodloader.event.impl.GuiEvent.Open(var1);
+      if (!xyz.openmodloader.OpenModLoader.INSTANCE.EVENT_BUS.post(event)) return; else var1 = event.getGui();
+//    OML END
       if(this.currentScreen != null) {
          this.currentScreen.onGuiClosed();
       }
@@ -904,6 +907,9 @@
       GlStateManager.enableTexture2D();
       this.mcProfiler.endSection();
       if(!this.skipRenderWorld) {
+//       OML START
+         xyz.openmodloader.OpenModLoader.INSTANCE.EVENT_BUS.post(new xyz.openmodloader.event.impl.UpdateEvent.RenderUpdate(this.timer.renderPartialTicks));
+//       OML END
          this.mcProfiler.endStartSection("gameRenderer");
          this.entityRenderer.updateCameraAndRender(this.timer.renderPartialTicks, var1);
          this.mcProfiler.endSection();
@@ -2454,7 +2460,8 @@
                if(var1 == this.gameSettings.keyBindFullscreen.getKeyCode()) {
                   this.toggleFullscreen();
                } else if(var1 == this.gameSettings.keyBindScreenshot.getKeyCode()) {
-                  this.ingameGUI.getChatGUI().printChatMessage(ScreenShotHelper.saveScreenshot(this.mcDataDir, this.displayWidth, this.displayHeight, this.framebufferMc));
+                  ITextComponent component = ScreenShotHelper.saveScreenshot(this.mcDataDir, this.displayWidth, this.displayHeight, this.framebufferMc);
+                  if (component != null) this.ingameGUI.getChatGUI().printChatMessage(component);
                }
             }
 
