--- /net/minecraft/client
+++ /net/minecraft/client
@@ -389,6 +389,7 @@
    }
 
    private void startGame() throws LWJGLException {
+      xyz.openmodloader.OpenModLoader.INSTANCE.minecraftConstruction();
       this.gameSettings = new GameSettings(this, this.mcDataDir);
       this.defaultResourcePacks.add(this.mcDefaultResourcePack);
       this.startTimerHackThread();
@@ -787,6 +788,8 @@
    }
 
    public void displayGuiScreen(@Nullable GuiScreen var1) {
+      xyz.openmodloader.event.impl.GuiEvent.Open event = new xyz.openmodloader.event.impl.GuiEvent.Open(var1);
+      if (!xyz.openmodloader.event.Events.OPEN_GUI.post(event)) return; else var1 = event.getGui();
       if(this.currentScreen != null) {
          this.currentScreen.onGuiClosed();
       }
@@ -904,6 +907,7 @@
       GlStateManager.enableTexture2D();
       this.mcProfiler.endSection();
       if(!this.skipRenderWorld) {
+         xyz.openmodloader.event.Events.RENDER_UPDATE.post(new xyz.openmodloader.event.impl.UpdateEvent.Render(this.timer.renderPartialTicks));
          this.mcProfiler.endStartSection("gameRenderer");
          this.entityRenderer.updateCameraAndRender(this.timer.renderPartialTicks, var1);
          this.mcProfiler.endSection();
@@ -2454,7 +2458,8 @@
                if(var1 == this.gameSettings.keyBindFullscreen.getKeyCode()) {
                   this.toggleFullscreen();
                } else if(var1 == this.gameSettings.keyBindScreenshot.getKeyCode()) {
-                  this.ingameGUI.getChatGUI().printChatMessage(ScreenShotHelper.saveScreenshot(this.mcDataDir, this.displayWidth, this.displayHeight, this.framebufferMc));
+                  ITextComponent component = ScreenShotHelper.saveScreenshot(this.mcDataDir, this.displayWidth, this.displayHeight, this.framebufferMc);
+                  if (component != null) this.ingameGUI.getChatGUI().printChatMessage(component);
                }
             }
 
