--- /net/minecraft/client
+++ /net/minecraft/client
@@ -389,6 +389,9 @@
    }
 
    private void startGame() throws LWJGLException {
+      //OME START
+      xyz.openmodloader.OpenModLoader.INSTANCE.minecraftConstruction();
+      //OME END
       this.gameSettings = new GameSettings(this, this.mcDataDir);
       this.defaultResourcePacks.add(this.mcDefaultResourcePack);
       this.startTimerHackThread();
