--- /net/minecraft/entity
+++ /net/minecraft/entity
@@ -28,8 +28,6 @@
 import net.minecraft.crash.ICrashReportDetail;
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.enchantment.EnchantmentProtection;
-import net.minecraft.entity.EntityList;
-import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.effect.EntityLightningBolt;
 import net.minecraft.entity.item.EntityBoat;
 import net.minecraft.entity.item.EntityItem;
@@ -308,6 +306,9 @@
    }
 
    public void onUpdate() {
+//    OML START
+      xyz.openmodloader.OpenModLoader.INSTANCE.EVENT_BUS.post(new xyz.openmodloader.event.impl.UpdateEvent.EntityUpdate(this));
+//    OML END
       if(!this.worldObj.isRemote) {
          this.setFlag(6, this.isGlowing());
       }
