--- /\net\minecraft\server\management
+++ /\net\minecraft\server\management
@@ -198,6 +198,9 @@
    }
 
    public boolean tryHarvestBlock(BlockPos var1) {
+      //OML START
+      if (!xyz.openmodloader.event.Events.BLOCK_DESTROY.post(new xyz.openmodloader.event.impl.BlockEvent.DestroyEvent(this.theWorld, this.theWorld.getBlockState(var1), var1))) return false;
+      //OML END
       if(this.gameType.isCreative() && this.thisPlayerMP.getHeldItemMainhand() != null && this.thisPlayerMP.getHeldItemMainhand().getItem() instanceof ItemSword) {
          return false;
       } else {
