--- /\net\minecraft\item
+++ /\net\minecraft\item
@@ -100,7 +100,7 @@
       return false;
    }
 
-   protected <T extends Comparable<T>> IBlockState makeState(IProperty<T> var1, Comparable<?> var2) {
-      return this.doubleSlab.getDefaultState().withProperty(var1, var2);
+   protected <T extends Comparable<T>> IBlockState makeState(IProperty<T> var1, Comparable<T> var2) {
+      return this.doubleSlab.getDefaultState().withProperty(var1,(T) var2);
    }
 }
