--- /net/minecraft/client/renderer/entity
+++ /net/minecraft/client/renderer/entity
@@ -37,7 +37,7 @@
    }
 
    protected <V extends EntityLivingBase, U extends LayerRenderer<V>> boolean addLayer(U var1) {
-      return this.layerRenderers.add(var1);
+      return this.layerRenderers.add((LayerRenderer<T>) var1);
    }
 
    protected <V extends EntityLivingBase, U extends LayerRenderer<V>> boolean removeLayer(U var1) {
