--- /net/minecraft/client/multiplayer
+++ /net/minecraft/client/multiplayer
@@ -11,7 +11,6 @@
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.audio.PositionedSoundRecord;
 import net.minecraft.client.entity.EntityPlayerSP;
-import net.minecraft.client.multiplayer.WorldClient;
 import net.minecraft.client.network.NetHandlerPlayClient;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.passive.EntityHorse;
@@ -106,6 +105,9 @@
          }
       }
 
+//    OML START
+      if (!xyz.openmodloader.OpenModLoader.INSTANCE.EVENT_BUS.post(new xyz.openmodloader.event.impl.BlockEvent.Destroy(this.mc.theWorld, this.mc.theWorld.getBlockState(var1), var1))) return false;
+//    OML END
       if(this.currentGameType.isCreative() && this.mc.thePlayer.getHeldItemMainhand() != null && this.mc.thePlayer.getHeldItemMainhand().getItem() instanceof ItemSword) {
          return false;
       } else {
@@ -161,7 +163,7 @@
       if(!this.mc.theWorld.getWorldBorder().contains(var1)) {
          return false;
       } else {
-         if(this.currentGameType.isCreative()) {
+         if(false) {
             this.connection.sendPacket(new CPacketPlayerDigging(CPacketPlayerDigging.a.START_DESTROY_BLOCK, var1, var2));
             clickBlockCreative(this.mc, this, var1, var2);
             this.blockHitDelay = 5;
@@ -173,11 +175,13 @@
             this.connection.sendPacket(new CPacketPlayerDigging(CPacketPlayerDigging.a.START_DESTROY_BLOCK, var1, var2));
             IBlockState var5 = this.mc.theWorld.getBlockState(var1);
             boolean var4 = var5.getMaterial() != Material.AIR;
+            if (!this.currentGameType.isCreative())
             if(var4 && this.curBlockDamageMP == 0.0F) {
                var5.getBlock().onBlockClicked(this.mc.theWorld, var1, this.mc.thePlayer);
             }
 
             if(var4 && var5.getPlayerRelativeBlockHardness(this.mc.thePlayer, this.mc.thePlayer.worldObj, var1) >= 1.0F) {
+               if (this.currentGameType.isCreative()) {clickBlockCreative(this.mc, this, var1, var2); this.blockHitDelay = 5;} else
                this.onPlayerDestroyBlock(var1);
             } else {
                this.isHittingBlock = true;
@@ -209,7 +213,7 @@
       if(this.blockHitDelay > 0) {
          --this.blockHitDelay;
          return true;
-      } else if(this.currentGameType.isCreative() && this.mc.theWorld.getWorldBorder().contains(var1)) {
+      } else if(false) {
          this.blockHitDelay = 5;
          this.connection.sendPacket(new CPacketPlayerDigging(CPacketPlayerDigging.a.START_DESTROY_BLOCK, var1, var2));
          clickBlockCreative(this.mc, this, var1, var2);
@@ -231,6 +235,7 @@
             if(this.curBlockDamageMP >= 1.0F) {
                this.isHittingBlock = false;
                this.connection.sendPacket(new CPacketPlayerDigging(CPacketPlayerDigging.a.STOP_DESTROY_BLOCK, var1, var2));
+               if (this.currentGameType.isCreative()) {clickBlockCreative(this.mc, this, var1, var2);} else
                this.onPlayerDestroyBlock(var1);
                this.curBlockDamageMP = 0.0F;
                this.stepSoundTickCounter = 0.0F;
