--- /\net\minecraft\client\settings
+++ /\net\minecraft\client\settings
@@ -464,7 +464,7 @@
          }
 
          this.soundLevels.clear();
-         List var1 = IOUtils.readLines(new FileInputStream(this.optionsFile));
+         List<String> var1 = IOUtils.readLines(new FileInputStream(this.optionsFile));
          NBTTagCompound var2 = new NBTTagCompound();
 
          for(String var4 : var1) {
