--- /net/minecraft/entity/ai
+++ /net/minecraft/entity/ai
@@ -22,7 +22,7 @@
       } else if(!this.theVillager.worldObj.isDaytime()) {
          return false;
       } else {
-         List var1 = this.theVillager.worldObj.getEntitiesWithinAABB(EntityIronGolem.class, this.theVillager.getEntityBoundingBox().expand(6.0D, 2.0D, 6.0D));
+         List<EntityIronGolem> var1 = this.theVillager.worldObj.getEntitiesWithinAABB(EntityIronGolem.class, this.theVillager.getEntityBoundingBox().expand(6.0D, 2.0D, 6.0D));
          if(var1.isEmpty()) {
             return false;
          } else {
