--- /net/minecraft/client/renderer/texture
+++ /net/minecraft/client/renderer/texture
@@ -134,7 +134,7 @@
 
       LOGGER.info("Created: {}x{} {}-atlas", new Object[]{Integer.valueOf(var3.getCurrentWidth()), Integer.valueOf(var3.getCurrentHeight()), this.basePath});
       TextureUtil.allocateTextureImpl(this.getGlTextureId(), this.mipmapLevels, var3.getCurrentWidth(), var3.getCurrentHeight());
-      HashMap var27 = Maps.newHashMap(this.mapRegisteredSprites);
+      HashMap<String, TextureAtlasSprite> var27 = Maps.newHashMap(this.mapRegisteredSprites);
 
       for(TextureAtlasSprite var30 : var3.getStichSlots()) {
          if(var30 == this.missingImage || this.generateMipmaps(var1, var30)) {
