--- /\net\minecraft\client\gui
+++ /\net\minecraft\client\gui
@@ -185,7 +185,10 @@
    private void addSingleplayerMultiplayerButtons(int var1, int var2) {
       this.buttonList.add(new GuiButton(1, this.width / 2 - 100, var1, I18n.format("menu.singleplayer", new Object[0])));
       this.buttonList.add(new GuiButton(2, this.width / 2 - 100, var1 + var2 * 1, I18n.format("menu.multiplayer", new Object[0])));
-      this.realmsButton = this.func_189646_b(new GuiButton(14, this.width / 2 - 100, var1 + var2 * 2, I18n.format("menu.online", new Object[0])));
+      //OML START
+      this.realmsButton = this.func_189646_b(new GuiButton(14, this.width / 2 + 2, var1 + var2 * 2, 98, 20, I18n.format("menu.online", new Object[0])));
+      this.func_189646_b(new GuiButton(21, this.width / 2 - 100, var1 + var2 * 2, 98, 20, "Open Mod Loader"));
+      //OML END
    }
 
    private void addDemoButtons(int var1, int var2) {
@@ -236,6 +239,11 @@
          }
       }
 
+      //OML START
+      if(var1.id == 21){
+         this.mc.displayGuiScreen(new xyz.openmodloader.client.gui.GuiOML(this));
+      }
+      //OML END
    }
 
    private void switchToRealms() {
@@ -438,6 +446,11 @@
          var8 = var8 + ("release".equalsIgnoreCase(this.mc.getVersionType())?"":"/" + this.mc.getVersionType());
       }
 
+      //OML START
+      for(String str : xyz.openmodloader.client.OpenModLoaderClient.getMainMenuStrings()){
+         this.drawString(this.fontRendererObj, str, 2, this.height - 20 - (xyz.openmodloader.client.OpenModLoaderClient.getMainMenuStrings().indexOf(str) * 10), -1);
+      }
+      //OML END
       this.drawString(this.fontRendererObj, var8, 2, this.height - 10, -1);
       String var9 = "Copyright Mojang AB. Do not distribute!";
       this.drawString(this.fontRendererObj, "Copyright Mojang AB. Do not distribute!", this.width - this.fontRendererObj.getStringWidth("Copyright Mojang AB. Do not distribute!") - 2, this.height - 10, -1);
