--- /net/minecraft/client/gui
+++ /net/minecraft/client/gui
@@ -91,6 +91,7 @@
             }
          }
 
+         xyz.openmodloader.event.Events.SPLASH_LOAD.post(new xyz.openmodloader.event.impl.SplashLoadEvent(var2));
          if(!var2.isEmpty()) {
             while(true) {
                this.splashText = (String)var2.get(RANDOM.nextInt(var2.size()));
@@ -185,7 +186,8 @@
    private void addSingleplayerMultiplayerButtons(int var1, int var2) {
       this.buttonList.add(new GuiButton(1, this.width / 2 - 100, var1, I18n.format("menu.singleplayer", new Object[0])));
       this.buttonList.add(new GuiButton(2, this.width / 2 - 100, var1 + var2 * 1, I18n.format("menu.multiplayer", new Object[0])));
-      this.realmsButton = this.func_189646_b(new GuiButton(14, this.width / 2 - 100, var1 + var2 * 2, I18n.format("menu.online", new Object[0])));
+      this.realmsButton = this.func_189646_b(new GuiButton(14, this.width / 2 + 2, var1 + var2 * 2, 98, 20, I18n.format("menu.online", new Object[0])));
+      this.func_189646_b(new GuiButton(21, this.width / 2 - 100, var1 + var2 * 2, 98, 20, "Open Mod Loader"));
    }
 
    private void addDemoButtons(int var1, int var2) {
@@ -236,6 +238,9 @@
          }
       }
 
+      if(var1.id == 21){
+         this.mc.displayGuiScreen(new xyz.openmodloader.client.gui.GuiOML(this));
+      }
    }
 
    private void switchToRealms() {
@@ -438,6 +443,9 @@
          var8 = var8 + ("release".equalsIgnoreCase(this.mc.getVersionType())?"":"/" + this.mc.getVersionType());
       }
 
+      for(String str : xyz.openmodloader.client.OpenModLoaderClient.getMainMenuStrings()){
+         this.drawString(this.fontRendererObj, str, 2, this.height - 20 - (xyz.openmodloader.client.OpenModLoaderClient.getMainMenuStrings().indexOf(str) * 10), -1);
+      }
       this.drawString(this.fontRendererObj, var8, 2, this.height - 10, -1);
       String var9 = "Copyright Mojang AB. Do not distribute!";
       this.drawString(this.fontRendererObj, "Copyright Mojang AB. Do not distribute!", this.width - this.fontRendererObj.getStringWidth("Copyright Mojang AB. Do not distribute!") - 2, this.height - 10, -1);
