--- /net/minecraft/client/gui
+++ /net/minecraft/client/gui
@@ -16,12 +16,6 @@
 import java.util.List;
 import java.util.Set;
 import net.minecraft.client.Minecraft;
-import net.minecraft.client.gui.FontRenderer;
-import net.minecraft.client.gui.Gui;
-import net.minecraft.client.gui.GuiButton;
-import net.minecraft.client.gui.GuiConfirmOpenLink;
-import net.minecraft.client.gui.GuiLabel;
-import net.minecraft.client.gui.GuiYesNoCallback;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.RenderHelper;
 import net.minecraft.client.renderer.RenderItem;
@@ -340,6 +334,9 @@
          for(int var4 = 0; var4 < this.buttonList.size(); ++var4) {
             GuiButton var5 = (GuiButton)this.buttonList.get(var4);
             if(var5.mousePressed(this.mc, var1, var2)) {
+//             OML START
+               if (!xyz.openmodloader.OpenModLoader.INSTANCE.EVENT_BUS.post(new xyz.openmodloader.event.impl.GuiEvent.ButtonClick(this, var5))) return;
+//             OML END
                this.selectedButton = var5;
                var5.playPressSound(this.mc.getSoundHandler());
                this.actionPerformed(var5);
@@ -370,6 +367,10 @@
       this.width = var2;
       this.height = var3;
       this.buttonList.clear();
+//    OML START
+      xyz.openmodloader.event.impl.GuiEvent.Init event = new xyz.openmodloader.event.impl.GuiEvent.Init(this, this.buttonList);
+      if (!xyz.openmodloader.OpenModLoader.INSTANCE.EVENT_BUS.post(event)) return; else this.buttonList = event.getButtonList();
+//    OML END
       this.initGui();
    }
 
