--- /net/minecraft/client/gui
+++ /net/minecraft/client/gui
@@ -340,6 +340,7 @@
          for(int var4 = 0; var4 < this.buttonList.size(); ++var4) {
             GuiButton var5 = (GuiButton)this.buttonList.get(var4);
             if(var5.mousePressed(this.mc, var1, var2)) {
+               if (!xyz.openmodloader.event.Events.BUTTON_CLICK.post(new xyz.openmodloader.event.impl.GuiEvent.ButtonClick(this, var5))) return;
                this.selectedButton = var5;
                var5.playPressSound(this.mc.getSoundHandler());
                this.actionPerformed(var5);
@@ -370,6 +371,8 @@
       this.width = var2;
       this.height = var3;
       this.buttonList.clear();
+      xyz.openmodloader.event.impl.GuiEvent.Init event = new xyz.openmodloader.event.impl.GuiEvent.Init(this, this.buttonList);
+      if (!xyz.openmodloader.event.Events.INIT_GUI.post(event)) return; else this.buttonList = event.getButtonList();
       this.initGui();
    }
 
